### DISCUSSION TITLE
Search function reveals filenames in password-protected folders.

### DISCUSSION THREAD

**Discussion by qinenergy** [spencerwooo/onedrive-vercel-index#735] [Q&A] [answered]
Thanks for the wonderful project.

Not sure if this is by design or not, but this behavior may not be what people expected from the password protection. I am not sure about the ideal logic here, maybe only include the files of a protected folder if the password for that folder is already provided before.

A quick example: search "mp4" in the live demo https://drive.swo.moe/ reveals that "watch this.mp4" exists in the private folder.
![image](https://user-images.githubusercontent.com/9915021/182387168-a5c04032-5bc4-4b97-8c9d-48b47a64a7e1.png)

**Comment by myl7** [comment, marked as answer]
Yes, password-protected files showing up in searching results is expected behavior. This is due to a caveat of OneDrive API (that is OneDrive business searching API results do not contain the path). More details can be found in #283.
If you are just looking for a visual defense, which is hide the results in the page though they are still accessible via some APIs, #648 may help.
